### DISCUSSION TITLE
Hangs when converting epub to pdf

### DISCUSSION THREAD

**Discussion by slmjkdbtl** [jgm/pandoc#10017] [Q&A]
Command:

```
pandoc -f epub -t pdf --pdf-engine=tectonic "book.epub" -o "book.pdf" --verbose
```

With the verbose option this command outputs the full document, ends at a `\end{document}`, but the program doesn't end and `book.pdf` isn't written.

```
pandoc 3.2.1
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: /Users/tga/.local/share/pandoc
Copyright (C) 2006-2024 John MacFarlane. Web: https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

(I only have tried tectonic as pdf engine and my current location and internet speed won't allow me to download and try latex)

**Comment by tarleb** [comment]
> (I only have tried tectonic as pdf engine and my current location and internet speed won't allow me to download and try latex)

That might be the issue: tectonic is a LaTeX engine that downloads packages on demand. It may be that it just hangs because it's trying to download the missing packages.

**Reply by slmjkdbtl** [reply]
Yes that seems to be exactly what happens, tried again and it works now, thanks!
